src/target/riscv/riscv.c

```diff
@@ -4087,9 +4106,8 @@ static int parse_ranges(struct list_head *ranges, const char *tcl_arg, const cha
 		high = MAX(high, low);
 
 		if (high > max_val) {
-			LOG_ERROR("Cannot expose %s register number %u, maximum allowed value is %u.", reg_type, high, max_val);
-			free(name);
-			free(args);
+			LOG_ERROR("Cannot expose %s register number %u, maximum allowed value is %u.",
+				reg_type, high, max_val);
```

Comment:
Now that hexadecimal literals are accepted, you may consider using hexadecimal literals also in the messages:

```suggestion
			LOG_ERROR("Cannot expose %s register number 0x%x, maximum allowed value is 0x%x.",
				reg_type, high, max_val);
```

Reply:
addressed